src/lex/mod.rs

```diff
@@ -865,7 +905,8 @@ impl Iterator for Lexer {
                         .with(LexError::UnknownToken(x as char))));
                 }
             };
-            self.seen_line_token |= data != Token::Hash;
+            // We've seen a token if this isn't # and this isn't whitespace
+            self.seen_line_token |= data != Token::Hash && !matches!(data, Token::Whitespace(_));
```

Comment:
This might be easier to read if you reversed the condition: `!(Token::Hash || whitespace)`.

Reply:
Done